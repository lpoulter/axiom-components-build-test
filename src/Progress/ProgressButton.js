import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Button from '../Button/Button';
import Cloak from '../Cloak/Cloak';
import ProgressInfinite from './ProgressInfinite';
import './ProgressButton.css';

const progressSizeMap = {
  small: '1rem',
  medium: '1.25rem',
  large: '1.5rem',
};

export default class ProgressButton extends Component {
  static propTypes = {
    /** Content to be inserted into the Button */
    children: PropTypes.node.isRequired,
    /** Flag if an action is in progress, shows the progress indicator */
    isInProgress: PropTypes.bool,
    /** Size of the Button. See Button[size]. */
    size: PropTypes.oneOf(['small', 'medium', 'large']),
  };

  render() {
    const {
      children,
      isInProgress,
      size = Button.defaultProps.size,
      ...rest
    } = this.props;

    const classes = classnames('ax-progress-button', {
      'ax-progress-button--in-progress': isInProgress,
    });

    return (
      <Button { ...rest }
          active={ isInProgress }
          size={ size }
          style="primary">
        <div className={ classes }>
          <Cloak
              className="ax-progress-button__content"
              invisible={ isInProgress }>
            { children }
          </Cloak>

          <Cloak
              className="ax-progress-button__indicator"
              invisible={ !isInProgress }>
            <ProgressInfinite
                color="white"
                sizeRem={ progressSizeMap[size] } />
          </Cloak>
        </div>
      </Button>
    );
  }
}
