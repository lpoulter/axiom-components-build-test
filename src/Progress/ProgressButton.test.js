import React from 'react';
import renderer from 'react-test-renderer';
import ProgressButton from './ProgressButton';

const getComponent = (props = {}) =>
  renderer.create(
    <ProgressButton { ...props }>Lorem</ProgressButton>
  );

describe('ProgressButton', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with isInProgress', () => {
    const component = getComponent({ isInProgress: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with size', () => {
    ['small', 'medium', 'large'].forEach((size) =>
      it(size, () => {
        const component = getComponent({ size });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      })
    );
  });
});
