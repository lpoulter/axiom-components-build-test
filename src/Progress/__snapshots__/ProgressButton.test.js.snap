// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressButton renders with defaultProps 1`] = `
<button
  className="ax-button ax-button--accent ax-button--primary ax-button--rectangle-medium"
>
  <div
    className="ax-progress-button"
  >
    <div
      className="ax-progress-button__content"
    >
      Lorem
    </div>
    <div
      className="ax-progress-button__indicator ax-cloak"
    >
      <svg
        className="ax-progress-infinite ax-progress-infinite--white ax-radial-progress ax-radial-progress--small"
        style={
          Object {
            "height": "1.25rem",
            "width": "1.25rem",
          }
        }
        viewBox="0 0 24 24"
      >
        <g
          transform="translate(12, 12) rotate(-90)"
        >
          <circle
            className="ax-progress-infinite__arc"
            r={10}
            style={
              Object {
                "stroke": "url(#ax-progress-infinite__filter--white)",
                "strokeDasharray": 62.83185307179586,
                "strokeDashoffset": 15.707963267948966,
                "strokeWidth": 4,
              }
            }
          />
        </g>
      </svg>
    </div>
  </div>
</button>
`;

exports[`ProgressButton renders with isInProgress 1`] = `
<button
  className="ax-button ax-button--accent ax-button--primary ax-button--rectangle-medium ax-button--active"
>
  <div
    className="ax-progress-button ax-progress-button--in-progress"
  >
    <div
      className="ax-progress-button__content ax-cloak"
    >
      Lorem
    </div>
    <div
      className="ax-progress-button__indicator ax-cloak ax-cloak--visible"
    >
      <svg
        className="ax-progress-infinite ax-progress-infinite--white ax-radial-progress ax-radial-progress--small"
        style={
          Object {
            "height": "1.25rem",
            "width": "1.25rem",
          }
        }
        viewBox="0 0 24 24"
      >
        <g
          transform="translate(12, 12) rotate(-90)"
        >
          <circle
            className="ax-progress-infinite__arc"
            r={10}
            style={
              Object {
                "stroke": "url(#ax-progress-infinite__filter--white)",
                "strokeDasharray": 62.83185307179586,
                "strokeDashoffset": 15.707963267948966,
                "strokeWidth": 4,
              }
            }
          />
        </g>
      </svg>
    </div>
  </div>
</button>
`;

exports[`ProgressButton renders with size large 1`] = `
<button
  className="ax-button ax-button--accent ax-button--primary ax-button--rectangle-large"
>
  <div
    className="ax-progress-button"
  >
    <div
      className="ax-progress-button__content"
    >
      Lorem
    </div>
    <div
      className="ax-progress-button__indicator ax-cloak"
    >
      <svg
        className="ax-progress-infinite ax-progress-infinite--white ax-radial-progress ax-radial-progress--small"
        style={
          Object {
            "height": "1.5rem",
            "width": "1.5rem",
          }
        }
        viewBox="0 0 24 24"
      >
        <g
          transform="translate(12, 12) rotate(-90)"
        >
          <circle
            className="ax-progress-infinite__arc"
            r={10}
            style={
              Object {
                "stroke": "url(#ax-progress-infinite__filter--white)",
                "strokeDasharray": 62.83185307179586,
                "strokeDashoffset": 15.707963267948966,
                "strokeWidth": 4,
              }
            }
          />
        </g>
      </svg>
    </div>
  </div>
</button>
`;

exports[`ProgressButton renders with size medium 1`] = `
<button
  className="ax-button ax-button--accent ax-button--primary ax-button--rectangle-medium"
>
  <div
    className="ax-progress-button"
  >
    <div
      className="ax-progress-button__content"
    >
      Lorem
    </div>
    <div
      className="ax-progress-button__indicator ax-cloak"
    >
      <svg
        className="ax-progress-infinite ax-progress-infinite--white ax-radial-progress ax-radial-progress--small"
        style={
          Object {
            "height": "1.25rem",
            "width": "1.25rem",
          }
        }
        viewBox="0 0 24 24"
      >
        <g
          transform="translate(12, 12) rotate(-90)"
        >
          <circle
            className="ax-progress-infinite__arc"
            r={10}
            style={
              Object {
                "stroke": "url(#ax-progress-infinite__filter--white)",
                "strokeDasharray": 62.83185307179586,
                "strokeDashoffset": 15.707963267948966,
                "strokeWidth": 4,
              }
            }
          />
        </g>
      </svg>
    </div>
  </div>
</button>
`;

exports[`ProgressButton renders with size small 1`] = `
<button
  className="ax-button ax-button--accent ax-button--primary ax-button--rectangle-small"
>
  <div
    className="ax-progress-button"
  >
    <div
      className="ax-progress-button__content"
    >
      Lorem
    </div>
    <div
      className="ax-progress-button__indicator ax-cloak"
    >
      <svg
        className="ax-progress-infinite ax-progress-infinite--white ax-radial-progress ax-radial-progress--small"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
        viewBox="0 0 24 24"
      >
        <g
          transform="translate(12, 12) rotate(-90)"
        >
          <circle
            className="ax-progress-infinite__arc"
            r={10}
            style={
              Object {
                "stroke": "url(#ax-progress-infinite__filter--white)",
                "strokeDasharray": 62.83185307179586,
                "strokeDashoffset": 15.707963267948966,
                "strokeWidth": 4,
              }
            }
          />
        </g>
      </svg>
    </div>
  </div>
</button>
`;
